engine/apps/grafana_plugin/views/self_hosted_install.py

```diff
@@ -1,43 +1,48 @@
-from django.apps import apps
 from django.conf import settings
 from rest_framework import status
-from rest_framework.authentication import get_authorization_header
 from rest_framework.request import Request
 from rest_framework.response import Response
 from rest_framework.views import APIView
 
-from apps.grafana_plugin.permissions import SelfHostedInvitationTokenVerified
-from apps.user_management.models import Organization
+from apps.grafana_plugin.helpers import GrafanaAPIClient
+from apps.user_management.models.organization import Organization, ProvisionedPlugin
 from apps.user_management.sync import sync_organization
 from common.api_helpers.mixins import GrafanaHeadersMixin
 
 
 class SelfHostedInstallView(GrafanaHeadersMixin, APIView):
-    permission_classes = (SelfHostedInvitationTokenVerified,)
-
-    def remove_invitation_token(self, token):
-        DynamicSetting = apps.get_model("base", "DynamicSetting")
-        self_hosted_settings = DynamicSetting.objects.get_or_create(
-            name="self_hosted_invitations",
-            defaults={
-                "json_value": {
-                    "keys": [],
-                }
-            },
-        )[0]
-        self_hosted_settings.json_value["keys"].remove(token)
-        self_hosted_settings.save(update_fields=["json_value"])
-
-    def post(self, request: Request) -> Response:
-        token_string = get_authorization_header(request).decode()
+    def post(self, _request: Request) -> Response:
```

Comment:
As permission_classes is removed should we check settings license here and return 403 if it is cloud configured?

Reply:
good idea! added this condition + test case